// @aws-cdk/aws-dms/lib/dms-canned-metrics.generated.js
"use strict";
// Copyright 2012-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DMSMetrics = void 0;
/* eslint-disable max-len */ // This is generated code - line lengths are difficult to control
class DMSMetrics {
    static cdcLatencyTargetSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCLatencyTarget',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cdcLatencySourceSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCLatencySource',
            dimensions,
            statistic: 'Sum',
        };
    }
    static availableMemoryAverage(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'AvailableMemory',
            dimensions,
            statistic: 'Average',
        };
    }
    static cdcChangesDiskTargetSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCChangesDiskTarget',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cdcChangesMemorySourceSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCChangesMemorySource',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cdcChangesMemoryTargetSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCChangesMemoryTarget',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cdcIncomingChangesSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCIncomingChanges',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cdcThroughputBandwidthSourceSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCThroughputBandwidthSource',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cdcThroughputBandwidthTargetSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCThroughputBandwidthTarget',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cdcThroughputRowsSourceSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCThroughputRowsSource',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cdcThroughputRowsTargetSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CDCThroughputRowsTarget',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cpuAllocatedSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CPUAllocated',
            dimensions,
            statistic: 'Sum',
        };
    }
    static cpuUtilizationAverage(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'CPUUtilization',
            dimensions,
            statistic: 'Average',
        };
    }
    static freeMemoryAverage(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'FreeMemory',
            dimensions,
            statistic: 'Average',
        };
    }
    static fullLoadThroughputBandwidthSourceSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'FullLoadThroughputBandwidthSource',
            dimensions,
            statistic: 'Sum',
        };
    }
    static fullLoadThroughputBandwidthTargetSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'FullLoadThroughputBandwidthTarget',
            dimensions,
            statistic: 'Sum',
        };
    }
    static fullLoadThroughputRowsSourceSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'FullLoadThroughputRowsSource',
            dimensions,
            statistic: 'Sum',
        };
    }
    static fullLoadThroughputRowsTargetSum(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'FullLoadThroughputRowsTarget',
            dimensions,
            statistic: 'Sum',
        };
    }
    static memoryAllocatedAverage(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'MemoryAllocated',
            dimensions,
            statistic: 'Average',
        };
    }
    static memoryUsageAverage(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'MemoryUsage',
            dimensions,
            statistic: 'Average',
        };
    }
    static swapUsageAverage(dimensions) {
        return {
            namespace: 'AWS/DMS',
            metricName: 'SwapUsage',
            dimensions,
            statistic: 'Average',
        };
    }
}
exports.DMSMetrics = DMSMetrics;
